src/main/java/faang/school/achievement/cache/AchievementCache.java

```diff
@@ -0,0 +1,35 @@
+package faang.school.achievement.cache;
+
+import faang.school.achievement.model.Achievement;
+import faang.school.achievement.repository.AchievementRepository;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class AchievementCache {
+    private final AchievementRepository achievementRepository;
+    private Map<String, Achievement> achievementsByName = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        List<Achievement> achievements = achievementRepository.findAll();
+        achievements.forEach(achievement -> {
+            achievementsByName.put(achievement.getTitle(), achievement);
+        });
+    }
+
+    public Optional<Achievement> getAchievement(String title) {
+        return Optional.ofNullable(achievementsByName.get(title));
+    }
+
+    public HashMap<String, Achievement> getAchievementsByName() {
+        return (HashMap<String, Achievement>) achievementsByName;
```

Comment:
I would return copy of hashmap then original. Reason -  Encapsulation / Immutability: return new HashMap<>(achievementsByName); Your hashmap here is like a cache it is bad practice if your cache it not immutable or threadsafe.
Or another may you could use concurrent hashmap

Reply:
Sounds reasonable